### PR TITLE
Adjust GitHub actions

### PR DESCRIPTION
**Pull request by PascalEgn**
ref: https://github.com/cern-sis/issues-scoap3/issues/133

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+on:
+  workflow_call:
+    inputs:
+      ref:
+        description: The reference to build
+        type: string
+        required: true
+
+jobs:
+  pytest:
```
Comment (MJedr) [inline .github/workflows/tests.yml]: can we give it another name, maybe `tests` for now

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Pre-Commit
+
+on:
+  workflow_call:
+    inputs:
+      ref:
+        description: The reference to build
+        type: string
+        required: true
+
```
Comment (MJedr) [inline .github/workflows/pre-commit.yml:10]: you can use one break line

File: .github/workflows/pull-request-main.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: Pull request main
+
+on:
+  pull_request_target:
+    branches: [main]
+    paths-ignore: [ "docs/**" ]
+
```
Comment (MJedr) [inline .github/workflows/pull-request-main.yml:7]: same

**Review by MJedr — COMMENTED** [review]
There are some things missing:
* we need to run tests on merge too
* we need to build an image and deploy it on merge, do you want to add this actions @PascalEgn?

**Comment by PascalEgn** [comment]
> * we need to run tests on merge too

Similar to [this](https://github.com/inspirehep/inspirehep/blob/master/.github/workflows/push-master.yml) right?

> * we need to build an image and deploy it on merge, do you want to add this actions @PascalEgn?

So after PR or merge, the image should get build, deployed and then used for the tests?

Anyways I will need help getting the Docker secrets into the repo

**Review by MJedr — APPROVED** [review]

**Pull request merged**